HKMP/Game/Client/Entity/MantisLord.cs

```diff
@@ -0,0 +1,907 @@
+
+            _subFsm.InsertMethod("Start", 0,
+                CreateUpdateMethod(() => { SendAnimationUpdate((byte) SubBattleAnimation.Start); }));
+
+            _subFsm.InsertMethod("Killed", 0,
+                CreateUpdateMethod(() => { SendAnimationUpdate((byte) SubBattleAnimation.Killed); }));
+        }
+
+        protected override void InternalInitializeAsSceneClient(byte? stateIndex) {
+            // Disable the decision FSM, this is handled by the host
+            RemoveAllTransitions(_subFsm);
+
+            if (stateIndex.HasValue) {
+                var state = (ThroneState) stateIndex;
+
+                if (state == ThroneState.Empty) {
+                    _subFsm.ExecuteActions("Init", 1, 2, 3);
+                    _subFsm.ExecuteActions("Start", 1);
+
+                    _subFsm.SetState("Init Pause");
+                }
+            }
+
+            // Now intialize the base class, as we might have activated its object
+            base.InternalInitializeAsSceneClient(stateIndex);
+        }
+
+        protected override void InternalSwitchToSceneHost() {
+            base.InternalSwitchToSceneHost();
+
+            // Reenable the decision FSM
+            RestoreAllTransitions(_subFsm);
+
+            // If the host leaves and the client has joined in the second phase, we need to trigger the actions again
+            if (_lastSubBattleAnimation == SubBattleAnimation.Idle && _subFsm.ActiveStateName == "Init Pause") {
+                _subFsm.SetState("Init Pause");
+            }
+
+            if (_lastSubBattleAnimation == SubBattleAnimation.Init) {
+                _subFsm.SetState("Start");
+            }
+
+            if (_lastSubBattleAnimation == SubBattleAnimation.Start) {
+                _subFsm.SetState("Set Subs");
+            }
+
+            if (_lastSubBattleAnimation == SubBattleAnimation.Killed) {
+                _subFsm.SetState("Killed");
+            }
+        }
+
+        public override void UpdateAnimation(byte animationIndex, byte[] animationInfo) {
+            if (animationIndex <= (byte) ThroneAnimation.Return || animationIndex == 255) {
+                base.UpdateAnimation(animationIndex, animationInfo);
+                return;
+            }
+
+            var animation = (SubBattleAnimation) animationIndex;
+
+            _lastSubBattleAnimation = animation;
+
+            if (animation == SubBattleAnimation.Init) {
+                _subFsm.ExecuteActions("Init", 1, 2, 3);
+            }
+
+            if (animation == SubBattleAnimation.Start) {
+                _subFsm.ExecuteActions("Start", 1);
+            }
+
+            if (animation == SubBattleAnimation.Killed) {
+                _subFsm.ExecuteActions("Killed", 1, 2);
+            }
+        }
+
+        private enum SubBattleAnimation {
+            Idle = ThroneAnimation.Return + 1,
+            Init,
+            Start,
+            Killed,
+        }
+    }
+
+    public class MantisLordS2 : MantisLordSub {
```

Comment:
Same as earlier comment about virtual method calls.
```suggestion
    public sealed class MantisLordS2 : MantisLordSub {
```